gcc/d/dfrontend/enum.c

```diff
@@ -517,22 +519,27 @@ void EnumMember::semantic(Scope *sc)
     if (errors || semanticRun >= PASSsemanticdone)
         return;
 
-    semanticRun = PASSsemantic;
     if (_scope)
         sc = _scope;
 
+    protection = ed->isAnonymous() ? ed->protection : Prot(PROTpublic);
+    storage_class = STCmanifest;
+    userAttribDecl = ed->isAnonymous() ? ed->userAttribDecl : NULL;
+
+    semanticRun = PASSsemantic;
+
     // The first enum member is special
     bool first = (this == (*ed->members)[0]);
 
-    if (type)
+    if (origType)
     {
-        type = type->semantic(loc, sc);
-        assert(value);          // "type id;" is not a valid enum member declaration
+        origType = origType->semantic(loc, sc);
+        type = origType;
```

Comment:
Turned out to be a merge problem, and not some hidden dependency (this line was missing, therefore `type == NULL`).